.github/scripts/create-vm.py

```diff
@@ -0,0 +1,465 @@
+import os
+import grpc
+import json
+import requests
+import math
+import logging
+import argparse
+import yaml
+import random
+import string
+from github import Github, Auth as GithubAuth
+from yandexcloud import SDK, RetryInterceptor, backoff_linear_with_jitter
+from yandex.cloud.compute.v1.instance_service_pb2_grpc import InstanceServiceStub
+from yandex.cloud.compute.v1.instance_pb2 import Instance
+from yandex.cloud.compute.v1.instance_service_pb2 import (
+    CreateInstanceRequest,
+    ResourcesSpec,
+    AttachedDiskSpec,
+    NetworkInterfaceSpec,
+    PrimaryAddressSpec,
+    OneToOneNatSpec,
+    DeleteInstanceRequest,
+    CreateInstanceMetadata,
+    DeleteInstanceMetadata,
+)
+from yandex.cloud.compute.v1.instance_pb2 import IpVersion
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s: %(levelname)s: %(message)s"
+)
+
+
+def github_output(key, value, secret=False):
+    GITHUB_OUTPUT = os.environ.get("GITHUB_OUTPUT")
+
+    if GITHUB_OUTPUT:
+        with open(GITHUB_OUTPUT, "a") as fp:
+            fp.write(f"{key}={value}\n")
+
+    logging.info(f"echo \"{key}={'******' if secret else value}\" >> $GITHUB_OUTPUT")
+
+
+def generate_github_label():
+    generated_string = "".join(
+        random.choices(string.ascii_lowercase + string.digits, k=8)
+    )
+    logging.info(f"Generated label: {generated_string}")
+    return generated_string
+
+
+class KeyValueAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):  # noqa: U100
+        kv_dict = {}
```

Comment:
Can be simplified to `kv_dict = dict([tuple(kv.split('=')) for kv in values.split(',')])`. But it is a bit more "pythonic" and not that readable as your version though.